diesel/src/sqlite/connection/raw.rs

```diff
@@ -62,27 +63,21 @@ impl RawConnection {
     }
 
     pub fn exec(&self, query: &str) -> QueryResult<()> {
-        let mut err_msg = ptr::null_mut();
         let query = CString::new(query)?;
         let callback_fn = None;
         let callback_arg = ptr::null_mut();
-        unsafe {
+        let result = unsafe {
             ffi::sqlite3_exec(
                 self.internal_connection.as_ptr(),
                 query.as_ptr(),
                 callback_fn,
                 callback_arg,
-                &mut err_msg,
-            );
-        }
+                ptr::null_mut(),
+            )
+        };
 
-        if err_msg.is_null() {
-            Ok(())
-        } else {
-            let msg = convert_to_string_and_free(err_msg);
-            let error_kind = DatabaseErrorKind::Unknown;
-            Err(DatabaseError(error_kind, Box::new(msg)))
-        }
+        ensure_sqlite_ok(result, self.internal_connection.as_ptr())?;
+        Ok(())
```

Comment:
```suggestion
        ensure_sqlite_ok(result, self.internal_connection.as_ptr())
```

Reply:
Thanks for the suggestion! I changed it around :)